### DISCUSSION TITLE
Options for pool light?

### DISCUSSION THREAD

**Discussion by Fumado** [tagyoureit/nodejs-poolController#634] [Q&A] [answered]
I have a standard pool light controlled with a 120v switch.  I'd rather not bring 120 volts into my NJSPC controller chassis for relay control, since everything in there is low voltage.  A smart switch that can be integrated into NJSPC would be ideal, but there doesn't seem to be any documentation on how to accomplish this.  

I setup a Home Assistant installation, but I'm not sure hot to make NJSPC see my HA devices, or if it even can see them.

Thanks for any suggestions/advice.

**Comment by rstrouse** [comment]
Is this a Nixie controller or do you have a Pentair panel?  Please tell me about the wattage on the light, whether you have a load center (a circuit breaker panel on the pad) and any hats that you have connected to your pi.  You have a few options here.

1. You can hook the relay to any relay on the pi that supports the amperage draw of the light.  Sequent makes a 40A relay that can switch a high wattage pool light if it is one of the traditional incandescent bulbs. https://sequentmicrosystems.com/collections/all-io-cards/products/three-40a-240v-relays-rs485-for-raspberry-pi
2. You can follow the instructions for high power relays.  This is the same method that is used by all of the commercial controllers on the market and is the most industrial installation.  The relays are typically installed in the high power section of the panel with the circuit breakers.  https://github.com/tagyoureit/nodejs-poolController/wiki/DIY-Standalone-Nixie-Pool-Controller#high-current-circuits
3.  I do not run Home Assistant so @Crewski may be able to give more insight into how to connect to a smart switch.  However, if that is an incandescent pool light be sure it supports the load.  It will eventually fail if not and run hot.  If it is LED it probably doesn't draw the 500-1000watts typically found on incandescent pool lights.

**Reply by Fumado** [reply]
This is the Nixie controller.  I don't know the wattage for the light.  I built the enclosure myself, and I have a touch screen that the pool guy operates.  I only have low voltage (5v for the Pi), and would prefer to keep it that way.  Switching out the standard light switch for a smart-switch is probably the easiest option for me.   Thanks.

**Comment by rock-crusher** [comment]
If it is just a matter of you not wanting to comingle the voltages inside your NJSPC controller then you could add something simple like this adjacent to the controller enclosure and connect your pool light to it.
This is available from Home Depot:

![Pump Start Relay Enclosure](https://user-images.githubusercontent.com/94593954/201542983-b7bf428c-56e9-4677-b9dd-0b4f98ca7501.JPG)

![Relay](https://user-images.githubusercontent.com/94593954/201543024-038edb20-429a-428d-bb18-b1f75b2aa9c9.JPG)

![Relay Specs](https://user-images.githubusercontent.com/94593954/201543038-71000c55-9c20-49c0-a06a-35de8fc9a85a.JPG)

**Reply by Telemachus1977** [reply]
Lots of cheap relays on the market that could easily switch your incandescent bulb, and you could just wire it up to some extra gpio pins on the pi and have a long wire that runs into your high voltage box.

**Comment by Crewski** [comment, marked as answer]
I would think the best/easiest route is to just replace your switch (I'm assuming its just a regular 15amp household toggle switch) with a smart switch of some sort.....but here are some options.

1. Install a smart switch in place of your current toggle switch.  This switch can be controlled via Home Assistant.  There are a million options out there, just keep in mind that most of them require internet access as they are not local control.  If you want local control via MQTT or ESPHome, look at the Shelley 1 or a switch with Tasmota/ESPHome on it already like this (https://www.amazon.com/TASMOTA-Martin-Jerry-Compatible-Devices/dp/B0B2RLKRJB/).  If you really wanted to use the njsPC dashboard for your control, you could make a light circuit in njsPC that you control via mqtt, then make and automation in Home Assistant to tie the switch and the fake light together.
2.  You can buy a cheap 5v relay module (https://www.amazon.com/HiLetgo-Channel-optocoupler-Support-Trigger/dp/B00LW15A4W/) to be controlled via the raspberry pi.  The module can be houses in you switch's electrical box, just replace the switch with a blank.  You would need to run 5v, GND, and a GPIO wire from the RPi to the box.  This would give you native control via njsPC.
3. Follow the instructions for high power relays like rstrouse suggested.

Since its just a regular bulb I'd go with option #1.  If later down the line you get an LED light that does different colors based on an on/off sequence, then #2 or #3 would be better.

**Reply by Telemachus1977** [reply]
#2 is cheap and reliable.  No need for a 3HP relay.

**Reply by Fumado** [reply]
The smart switch is the best option.  It requires a lot more effort, but it keeps things separate.
